lib/vector-core/src/transform/config.rs

```diff
@@ -78,7 +78,11 @@ pub trait TransformConfig: core::fmt::Debug + Send + Sync + dyn_clone::DynClone
 
     fn input(&self) -> Input;
 
-    fn outputs(&self) -> Vec<Output>;
+    /// Returns a list of outputs to which this transform can deliver events.
+    ///
+    /// The provided `merged_definition` can be used by transforms to understand the expected shape
+    /// of events flowing through the transform.
+    fn outputs(&self, merged_definition: &schema::Definition) -> Vec<Output>;
```

Comment:
This is _somewhat_ unfortunate, as it requires stub data to be passed into a lot of places that have no business knowing or caring about schemas.

In the initial spike a few months ago, I had a separate `fn schema_definition(context: TransformContext) -> Option<schema::Definition>` added to this trait, which prevents the churn this change created, but since the introduction of multiple outputs, that approach seemed brittle to me, as you (as a developer) had to guarantee that whatever list of schemas you returned in that new method, they had to be linked to an existing `Output` defined in the unmodified `fn outputs()` method.

All in all, I think this gives us the best guarantee that people follow the required API contract, albeit with a bit more churn. Happy to hear other thoughts on potential solutions though.